components/Tabs/Tabs.js

```diff
@@ -1,58 +1,63 @@
 class TabLink {
-  constructor(element){
+  constructor(element) {
     // assign this.element to the element reference
-    this.element;
+    this.element = element;
     // Get the tab data attribute and save the value here
-    this.tabData = ; 
+    this.tabData = this.element.dataset.tab;
     // Find all elements with the .card class in index.html that correspond to the tab data attribute
+    this.cards = document.querySelectorAll(".card");
     // If the data is 'all' then select all cards regardless of their data attribute
-    if(){
-      this.cards = ;
+    if (this.tabData === "all") {
+      this.cards = document.querySelectorAll(".card");
     } else {
-      this.cards = ;
+      this.cards = document.querySelectorAll(`.card[data-tab="${this.tabData}`);
     }
 
-    // Map over the cards array and convert each card element into a new instance of the TabCard class. 
+    // Map over the cards array and convert each card element into a new instance of the TabCard class.
     // Pass in the card object to the TabCard class.
-    this.cards = Array.from(this.cards).map();
+    this.cards = Array.from(this.cards).map(linkItem => new TabCard(linkItem));
     // Add a click event that invokes selectTab
-    this.element.addEventListener();
+    this.element.addEventListener("click", () => {
+      this.selectTab();
+    });
   }
 
-  selectTab(){
-
+  selectTab() {
```

Comment:
This is good! But, the 'all' tab stays selected (white background) even when other tabs are clicked. They do filter properly though.